src/litserve/loops.py

```diff
@@ -208,9 +208,14 @@ def run_batched_loop(
             callback_runner.trigger_event(EventTypes.BEFORE_PREDICT, lit_api=lit_api)
             y = _inject_context(contexts, lit_api.predict, x)
             callback_runner.trigger_event(EventTypes.BEFORE_PREDICT, lit_api=lit_api)
+            if isinstance(y, str):
+                logger.warning(
+                    "The 'predict' method returned a string instead of a list of predictions. "
+                    "When batching is enabled, 'predict' should return a list of predictions. "
+                    "To avoid unexpected behavior, ensure 'predict' returns a list of predictions, or implement 'LitAPI.unbatch' correctly."
+                )
```

Comment:
The cleanest way to do this is to put it [here](https://github.com/Lightning-AI/LitServe/blob/e4d6b4aab221c3cc93daa49b413700e6d2d6a6ef/src/litserve/api.py#L66)

Reply:
I had something there in the beginning actually, will do it! Thanks.